ml-agents/mlagents/trainers/tests/test_trainer_controller.py

```diff
@@ -3,13 +3,17 @@
 
 from mlagents.tf_utils import tf
 from mlagents.trainers.trainer_controller import TrainerController
+from mlagents.trainers.ghost.controller import GhostController
 from mlagents.trainers.sampler_class import SamplerManager
 
+TRAINER_FACTORY = MagicMock()
```

Comment:
@ervteng is there a better way to mock this out?

Reply:
I'd say slightly cleaner to do in the functions where it's used instead of globally.

Reply:
Done.